components/string.rst

```diff
@@ -0,0 +1,451 @@
+    (new UnicodeString($word))->length();   // 4 (graphemes)
+
+    // some symbols require double the width of others to represent them when using
+    // a monospaced font (e.g. in a console). This method returns the total width
+    // needed to represent the entire word
+    $word = 'नमस्ते';
+    (new ByteString($word))->width();      // 18
+    (new CodePointString($word))->width(); // 4
+    (new UnicodeString($word))->width();   // 4
+
+    // only returns TRUE if the string is exactly an empty string (not even white spaces)
+    u('hello world')->isEmpty();  // false
+    u('     ')->isEmpty();        // false
+    u('')->isEmpty();             // true
+
+    // removes all white spaces from the start and end of the string and replaces two
+    // or more consecutive white spaces inside contents by a single white space
+    u("  \n\n   hello        world \n    \n")->collapseWhitespace(); // 'hello world'
+
+Methods to Change Case
+~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // changes all graphemes/code points to lower case
+    u('FOO Bar')->lower();  // 'foo bar'
+
+    // when dealing with different languages, uppercase/lowercase is not enough
+    // there are three cases (lower, upper, title), some characters have no case,
+    // case is context-sensitive and locale-sensitive, etc.
+    // this method returns a string that you can use in case-insensitive comparisons
+    u('FOO Bar')->folded();             // 'foo bar'
+    u('Die O\'Brian Straße')->folded(); // "die o'brian strasse"
+
+    // changes all graphemes/code points to upper case
+    u('foo BAR')->upper(); // 'FOO BAR'
+
+    // changes all graphemes/code points to "title case"
+    u('foo bar')->title();     // 'Foo bar'
+    u('foo bar')->title(true); // 'Foo Bar'
+
+    // changes all graphemes/code points to camelCase
+    u('Foo: Bar-baz.')->camel(); // 'fooBarBaz'
+    // changes all graphemes/code points to snake_case
+    u('Foo: Bar-baz.')->snake(); // 'foo_bar_baz'
+    // other cases can be achieved by chaining methods. E.g. PascalCase:
+    u('Foo: Bar-baz.')->camel()->title(); // 'FooBarBaz'
+
+The methods of all string classes are case sensitive by default. You can perform
+case-insensitive operations with the ``ignoreCase()`` method::
+
+    u('abc')->indexOf('B');               //  null
+    u('abc')->ignoreCase()->indexOf('B'); // 1
+
+Methods to Append and Prepend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // adds the given content (one or more strings) at the beginning/end of the string
+    u('world')->prepend('hello');      // 'helloworld'
+    u('world')->prepend('hello', ' '); // 'hello world'
+
+    u('hello')->append('world');      // 'helloworld'
+    u('hello')->append(' ', 'world'); // 'hello world'
+
+    // adds the given content at the beginning/end of the string, but only if not present
+    u('Name')->ensureStart('get');    // 'getName'
+    u('getName')->ensureStart('get'); // 'getName'
```

Comment:
I would add this case:

```php
u('getgetName')->ensureStart('get'); // 'getName'
```

(btw: I wouldn't expect this behavior from the method name. Is this a common behavior for such a method?)

Reply:
Very good catch! I added a new example. About the behavior, I'm not 100% convinced ... but I'd say it's the best possible choice.